hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -367,82 +305,53 @@ public void onStateInitialized(
-        this.platform = platform;
-
-        //noinspection ConstantValue
-        assert platformState != null : "Platform should never pass a null platform state";
+        this.configProvider = new ConfigProviderImpl(trigger == GENESIS, metrics);
+        logger.info(
+                "Initializing Hedera state version {} in {} mode with trigger {} and previous version {}",
+                version,
+                trigger,
+                configProvider
+                        .getConfiguration()
+                        .getConfigData(HederaConfig.class)
+                        .activeProfile(),
+                previousVersion == null ? "<NONE>" : previousVersion);
         logger.info(
-                "Initializing Hedera state with trigger {} and previous version {} instance of {}",
-                () -> trigger,
-                () -> previousVersion == null ? "<NONE>" : previousVersion,
-                () -> previousVersion == null
-                        ? "<NONE>"
-                        : previousVersion.getClass().getName());
+                "Platform state includes freeze time={} and last frozen={}",
+                platformState.getFreezeTime(),
+                platformState.getLastFrozenTime());
 
+        HederaSoftwareVersion deserializedVersion = null;
         // We do not support downgrading from one version to an older version.
-        final HederaSoftwareVersion deserializedVersion;
-        if (previousVersion instanceof HederaSoftwareVersion) {
-            deserializedVersion = (HederaSoftwareVersion) previousVersion;
-            if (isDowngrade(version, deserializedVersion)) {
+        if (previousVersion instanceof HederaSoftwareVersion hederaSoftwareVersion) {
+            deserializedVersion = hederaSoftwareVersion;
+            if (version.isBefore(deserializedVersion)) {
                 logger.fatal(
                         "Fatal error, state source version {} is higher than node software version {}",
                         deserializedVersion,
                         version);
-                System.exit(1);
+                throw new IllegalStateException("Cannot downgrade from " + deserializedVersion + " to " + version);
             }
-        } else if (previousVersion instanceof SerializableSemVers) {
-            deserializedVersion = new HederaSoftwareVersion(
-                    toPbj(((SerializableSemVers) previousVersion).getProto()),
-                    toPbj(((SerializableSemVers) previousVersion).getServices()),
-                    0);
         } else {
-            deserializedVersion = new HederaSoftwareVersion(SemanticVersion.DEFAULT, SemanticVersion.DEFAULT, 0);
+            if (previousVersion != null) {
+                logger.fatal("Deserialized state not created with Hedera software");
+                throw new IllegalStateException("Deserialized state not created with Hedera software");
+            }
```

Comment:
So, we don't allow migrating from mono-state now?

Reply:
True Neeha, this PR begins the mono removal that I am continuing in [this](https://github.com/hashgraph/hedera-services/pull/13828) draft PR.